execute/plugin_functions.go

```diff
@@ -621,58 +757,57 @@ func computeNoncesConsensus(
 	return consensusNonces
 }
 
-// getConsensusObservation merges all attributed observations into a single observation based on which values have
-// consensus among the observers.
-func getConsensusObservation(
+// computeConsensusObservation aggregates multiple attributed observations to produce a single consensus observation.
+// The provided f is required for computing the consensus on fChain prior to computing the observation consensus.
+func computeConsensusObservation(
 	lggr logger.Logger,
-	aos []plugincommon.AttributedObservation[exectypes.Observation],
-	destChainSelector cciptypes.ChainSelector,
-	F int,
+	observations []plugincommon.AttributedObservation[exectypes.Observation],
 	destChain cciptypes.ChainSelector,
+	f int,
 ) (exectypes.Observation, error) {
-	observedFChains := make(map[cciptypes.ChainSelector][]int)
-	for _, ao := range aos {
-		obs := ao.Observation
-		for chain, f := range obs.FChain {
-			observedFChains[chain] = append(observedFChains[chain], f)
-		}
-	}
-	// consensus on the fChain map uses the role DON F value
-	// because all nodes can observe the home chain.
-	donThresh := consensus.MakeConstantThreshold[cciptypes.ChainSelector](consensus.TwoFPlus1(F))
-	fChain := consensus.GetConsensusMap(lggr, "fChain", observedFChains, donThresh)
-	_, ok := fChain[destChain] // check if the destination chain is in the FChain.
+	fChain := getConsensusFChain(lggr, observations, f)
+
+	destFChain, ok := fChain[destChain]
 	if !ok {
-		return exectypes.Observation{}, fmt.Errorf("destination chain %d is not in FChain", destChain)
+		return exectypes.Observation{}, fmt.Errorf("destination chain %d is not in FChain %v", destChain, fChain)
 	}
 
-	lggr.Debugw("getConsensusObservation decoded observations", "aos", aos)
-
-	mergedCommitObservations := mergeCommitObservations(lggr, aos, fChain)
-
-	lggr.Debugw("merged commit observations", "mergedCommitObservations", mergedCommitObservations)
-
-	mergedMessageObservations := mergeMessageObservations(lggr, aos, fChain)
-	lggr.Debugw("merged message observations", "mergedMessageObservations", mergedMessageObservations)
-
-	mergedTokenObservations := mergeTokenObservations(lggr, aos, fChain)
+	consensusObservation := exectypes.NewObservation(
+		computeCommitObservationsConsensus(lggr, observations, fChain),
+		computeMessageObservationsConsensus(lggr, observations, fChain),
+		computeTokenDataObservationsConsensus(lggr, observations, fChain),
+		computeNoncesConsensus(lggr, observations, destFChain),
+		dt.Observation{},
+		computeMessageHashesConsensus(lggr, observations, fChain),
+	)
 
-	lggr.Debugw("merged token data observations", "mergedTokenObservations", mergedTokenObservations)
+	lggr.Debugw("computeConsensusObservation has finished computing the consensus observation",
+		"fChain", fChain,
+		"observations", observations,
```

Comment:
Logging all the observations might lead to the heavy impact on the underlying Loki engine. Afaik we also trim logs longer than X bytes (not sure what's the thresholds) in Loki. Maybe instead of logging everything, we should only print chains and merkle roots here to limit the size?